src/Illuminate/Database/Eloquent/Relations/Relation.php

```diff
@@ -273,6 +280,21 @@ public function wrap($value)
     }
 
     /**
+     * Set the morph map for polymorphic relations.
+     * 
+     * @param  array|null $map
+     * @return array
```

Comment:
Either this needs to be `array|null` or you have to return the map in every case.